stdlib/ballerina-http/src/main/ballerina/ballerina/net/http/caching.bal

```diff
@@ -0,0 +1,329 @@
+
+    foreach directive in directives {
+        directive = directive.trim();
+        if (directive == NO_CACHE) {
+            request.cacheControl.noCache = true;
+        } else if (directive == NO_STORE) {
+            request.cacheControl.noStore = true;
+        } else if (directive == NO_TRANSFORM) {
+            request.cacheControl.noTransform = true;
+        } else if (directive == ONLY_IF_CACHED) {
+            request.cacheControl.onlyIfCached = true;
+        } else if (directive.hasPrefix(MAX_AGE)) {
+            request.cacheControl.maxAge = getDirectiveValue(directive);
+        } else if (directive == MAX_STALE) {
+            request.cacheControl.maxStale = MAX_STALE_ANY_AGE;
+        } else if (directive.hasPrefix(MAX_STALE)) {
+            request.cacheControl.maxStale = getDirectiveValue(directive);
+        } else if (directive.hasPrefix(MIN_FRESH)) {
+            request.cacheControl.minFresh = getDirectiveValue(directive);
+        }
+        // non-standard directives are ignored
+    }
+}
+
+function <OutRequest request> parseOutReqCacheControlHeader () {
+    string cacheControl = request.getHeader(CACHE_CONTROL);
+    request.cacheControl = {};
+
+    // If the request doesn't contain a cache-control header, resort to default cache control settings
+    if (cacheControl == null) {
+        return;
+    }
+
+    string[] directives = cacheControl.split(",");
+
+    foreach directive in directives {
+        directive = directive.trim();
+        if (directive == NO_CACHE) {
+            request.cacheControl.noCache = true;
+        } else if (directive == NO_STORE) {
+            request.cacheControl.noStore = true;
+        } else if (directive == NO_TRANSFORM) {
+            request.cacheControl.noTransform = true;
+        } else if (directive == ONLY_IF_CACHED) {
+            request.cacheControl.onlyIfCached = true;
+        } else if (directive.hasPrefix(MAX_AGE)) {
+            request.cacheControl.maxAge = getDirectiveValue(directive);
+        } else if (directive == MAX_STALE) {
+            request.cacheControl.maxStale = MAX_STALE_ANY_AGE;
+        } else if (directive.hasPrefix(MAX_STALE)) {
+            request.cacheControl.maxStale = getDirectiveValue(directive);
+        } else if (directive.hasPrefix(MIN_FRESH)) {
+            request.cacheControl.minFresh = getDirectiveValue(directive);
+        }
+        // non-standard directives are ignored
+    }
+}
+
+function appendFields (string[] fields) (string) {
+    if (fields != null && lengthof fields > 0) {
+        return "=\"" + buildCommaSeparatedString(fields) + "\"";
+    }
+
+    return "";
+}
+
+function buildCommaSeparatedString (string[] values) (string) {
+    string delimitedValues = values[0];
+
+    int i = 1;
+    while (i < lengthof values) {
+        delimitedValues = delimitedValues + ", " + values[i];
+        i = i + 1;
+    }
+
+    return delimitedValues;
+}
+
+function getDirectiveValue (string directive) (int) {
```

Comment:
Shall we rename this method to getCacheExpirationDirectiveValue(..) as we don't pass all the caching directive to this function??